python/ray/util/metrics.py

```diff
@@ -121,6 +121,13 @@ def _record(
         self._validate_tags(final_tags)
         self._metric.record(value, tags=final_tags)
 
+    def inc_set_or_observe(
```

Comment:
It looks weird to have a method that does completely different things in each subclass. 
Can we just check the type on the stats.py side?

Reply:
+1 

Let's make sure we call appropriate methods for every metric type (set for Gauge, inc for Counter, etc)